feat(x509): add -certopt ext_oid to show OIDs for known extensions

ATIC-Yugandhar opened the pull request:
Implement the ability to display the Object Identifier (OID) alongside the human-readable name for known X.509 extensions when using the 'openssl x509 -text' command. This is controlled by the new '-certopt ext_oid' sub-option.

Includes updates to the command's documentation and adds an automated test case to verify the functionality.

Fixes #27182
<!--
Thank you for your pull request. Please review these requirements:

Contributors guide: https://github.com/openssl/openssl/blob/master/CONTRIBUTING.md

Other than that, provide a description above this comment if there isn't one already

If this fixes a GitHub issue, make sure to have a line saying 'Fixes #XXXX' (without quotes) in the commit message.
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
- [x] documentation is added or updated
- [x] tests are added or updated

ATIC-Yugandhar committed:
feat(x509): add -certopt ext_oid to show OIDs for known extensions

Implement the ability to display the Object Identifier (OID) alongside
the human-readable name for known X.509 extensions when using the
'openssl x509 -text' command. This is controlled by the new
'-certopt ext_oid' sub-option.

Includes updates to the command's documentation and adds an
automated test case to verify the functionality.

Fixes #27182

ATIC-Yugandhar committed:
refactor code

ATIC-Yugandhar committed:
Add test case for -certopt ext_oid in x509 -text output

This commit adds a subtest to 25-test_x509.t to verify the -certopt ext_oid
option, which displays OIDs for known certificate extensions in the x509
-text output. The test checks that OIDs are absent by default and present
when ext_oid is specified, using rootcert.pem as the test certificate.
Direct command execution is used to ensure reliable output capture.

Fixes #27182

ATIC-Yugandhar committed:
Refactor test case for #27182 resolution

This commit refactors the test case in 25-test_x509.t to validate the
-certopt ext_oid option, ensuring OIDs for known certificate extensions
are correctly displayed in the x509 -text output. The updated test uses
rootcert.pem to confirm OIDs are absent by default and present when
ext_oid is specified, leveraging direct command execution for reliable
output capture.

Resolves #27182

t8m commented:
This definitely cannot be accepted with CLA: trivial, please remove that commit message annotation and submit a regular CLA.

Also please drop the merge commit as we cannot merge PRs with merge commits.

ATIC-Yugandhar commented:
Hi @t8m , thanks for the feedback!

I've removed the `CLA: trivial` annotation from the commit message and rebased the branch to remove the merge commit.

Also, I've submitted my signed Individual CLA to `cla@openssl.org`.

Please let me know if anything else is needed. Happy to update further if required!

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 30 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 92 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 123 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 185 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 216 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 61 days ago

beldmit commented on apps/x509.c:
```diff
@@ -100,8 +99,7 @@ const OPTIONS x509_options[] = {
     {"issuer", OPT_ISSUER, '-', "Print issuer DN"},
-    {"nameopt", OPT_NAMEOPT, 's',
-     "Certificate subject/issuer name printing options"},
+    {"nameopt", OPT_NAMEOPT, 's', "Certificate subject/issuer name printing options"},
```
This change seems unnecessary

beldmit commented on include/openssl/x509.h.in:
```diff
@@ -67,6 +67,9 @@ extern "C" {
 
+/* X509v3 extension flags */
+# define X509_FLAG_EXT_OID (1UL << 31)
+
```
Do we need a global const here? Can't it be local for apps/x509.c at least for now?

beldmit reviewed (changes requested):
Unfortunately, this PR need rebase after code reformatting, and also I'd adjust some changes

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 30 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 61 days ago

openssl-machine commented:
This PR has been closed.  It was waiting for the creator to make requested changes but it has not been updated for 90 days.
